### DISCUSSION TITLE
Collection search rest endpoint conflict

### DISCUSSION THREAD

**Discussion by ccancellieri** [stac-utils/stac-fastapi#751] [Q&A]
Hi I'm opening this 'issue' to discuss about collection search.
Here is a working proposal but I've had to change the endpoint.
https://github.com/stac-utils/stac-fastapi-elasticsearch-opensearch/pull/298

I'm encountering the following problems:

We still need to use the POST to create new Collection(s) but the incoming expected body is different
Incompatible with Transactional (how to have it working?)

Questions:

Why the specifications define a new endpoint to search for items: .../search while for the collections we are overloading the /collections ?
Do you have any working example about having both search and creation working at the same time?

**Comment by vincentsarago** [comment]
Hi @ccancellieri 

This was already discussed in https://github.com/stac-utils/stac-fastapi/pull/736 and https://github.com/stac-utils/stac-fastapi/pull/739 

stac-fastapi itself allow you to create both, but you can't have both in the same application 🤷 IMO this is an issue at the specification level

In theory we could have a `POST - /Collections` endpoint which support both the search and the transaction but both are separate extension which are not linked which is why we (in the first iteration) could not propose a `unify` solution. That's said you could create such endpoint in your own application.

Worth noting that both `POST - /Collections` for search and for transaction should required different auth system (I don't think you'll want everyone to be able to create new collection, but you want everyone to search for them) which in a case of a unify solution will make the code base quite a mess (if we, at stac-fastapi level, try to support this)

**Reply by m-mohr** [reply]
This is not really a specification issue. POST to /collections for Collection search is a proprietary extension only available in stac-fastapi

 In my opinion the POST for Collection Search should be optional for providers to opt-in/out so that both extensions can be used alongside and according to the specs. As far as I understand #739 that might already the case?

**Comment by ccancellieri** [comment]
I'm going to discuss this over the spec repo as it looks a clear overlap:
https://github.com/stac-api-extensions/collection-search/issues/17
